ks3-jdk/src/jvmMain/kotlin/io/ks3/java/math/BigDecimalSerializers.kt

```diff
@@ -1,3 +1,67 @@
 package io.ks3.java.math
 
-// TODO
+import io.ks3.standard.doubleSerializer
+import io.ks3.standard.stringSerializer
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonEncoder
+import kotlinx.serialization.json.JsonUnquotedLiteral
+import kotlinx.serialization.json.jsonPrimitive
+import java.math.BigDecimal
+
+/**
+ * Encodes a [BigDecimal] as a string, preserving the full precision of the number. Wraps the number in quotes.
+ */
+typealias BigDecimalAsString = @Serializable(with = BigDecimalAsStringSerializer::class) BigDecimal
+
+/**
+ * Encodes a [BigDecimal] to it's double representation. [Double] have limited precision, so the number might not be
+ * exactly the same as the original [BigDecimal].
+ */
+typealias BigDecimalAsDouble = @Serializable(with = BigDecimalAsDoubleSerializer::class) BigDecimal
+
+/**
+ * Encodes a [BigDecimal] as an unquoted JSON literal, preserving the full precision of the number while being encoded as a number.
+ *
+ * Note that this typealias is primarily meant for JSON, other formats will have the [BigDecimal] encoded as a string.
+ */
+@ExperimentalSerializationApi
+typealias BigDecimalAsJsonLiteral = @Serializable(with = BigDecimalAsJsonLiteralSerializer::class) BigDecimal
```

Comment:
`BigDecimalAsJsonLiteral` -> `JsonLiteral` is the technical name of the KxS tool used to encode it, so the name is accurate, but I suspect it's not very clear and doesn't make the intended usage clear.

I've got some other suggestions:

* `BigDecimalAsJsonNumber` (my favourite)
* `BigDecimalAsMaxPrecisionJsonNumber`
* `BigDecimalAsExactJsonNumber`
* `BigDecimalAsPreciseJsonNumber`

'Json' could be dropped from the name too, because if another format has a special way of encoding BigDecimals then the typealias could be reused. But they're only typealiases, I think it's fine to be specific for now.

* `BigDecimalAsExactNumber`
* `BigDecimalWithFullPrecision`

Reply:
`BigDecimalAsJsonNumber` does properly indicate the [semantics involved](https://www.rfc-editor.org/rfc/rfc7159#section-6).

Besides, strings are literals too and the `JsonLiteral` has been renamed in kotlinx serialization to [JsonPrimitive](https://github.com/Kotlin/kotlinx.serialization/blob/84d4af04a688286ad1b51a72130ccda691caff75/formats/json/commonMain/src/kotlinx/serialization/json/JsonElement.kt#L143).